app/lib/frontend/handlers.dart

```diff
@@ -169,11 +175,15 @@ Future<shelf.Response> searchHandler(shelf.Request request) async {
 
   final resultPage = await searchService.search(query, useService);
   final links = new SearchLinks(query, resultPage.totalCount);
+  final String content = templateService.renderSearchPage(resultPage, links);
+
   _searchBackendTracker.add(resultPage.backend);
   _searchBackendLatencyTracker.add(resultPage.latency);
   _searchOverallLatencyTracker.add(sw.elapsed);
 
-  return htmlResponse(templateService.renderSearchPage(resultPage, links));
+  // do not await on memcache update
+  searchMemcache?.setUiSearchPage(cacheUrl, content);
```

Comment:
Updated with await. Is there any timeout in the `memcache` client that will make sure we don't hang with an otherwise prepared response page?